repositories/src/test/scala/cromwell/pipeline/datastorage/dao/mongo/DocumentRepositoryTest.scala

```diff
@@ -1,17 +1,44 @@
 package cromwell.pipeline.datastorage.dao.mongo
 
+import com.dimafeng.testcontainers.MongoDBContainer
+import cromwell.pipeline.database.TestMongoEngine
+import cromwell.pipeline.datastorage.dao.repository.ProjectConfigurationRepository
+import cromwell.pipeline.datastorage.dao.utils.TestProjectUtils.getDummyProjectConfiguration
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.utils.TestContainersUtils.getMongoContainer
+import cromwell.pipeline.utils.{ MongoConfig, TestContainersUtils }
 import org.mockito.Mockito.when
 import org.mongodb.scala.result.UpdateResult
 import org.mongodb.scala.{ Document, MongoCollection }
 import org.scalatest.{ AsyncWordSpec, Matchers }
 import org.scalatestplus.mockito.MockitoSugar
 
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext }
+
 class DocumentRepositoryTest extends AsyncWordSpec with Matchers with MockitoSugar {
 
-  private val updateResult = mock[UpdateResult]
-  private val collection = mock[MongoCollection[Document]]
+  val container: MongoDBContainer = getMongoContainer()
+
+  override implicit val executionContext: ExecutionContext = ExecutionContext.global
```

Comment:
Why do you override test execution context with the global one?

Reply:
There was a problem with lock in first test but it's resolved now